internal/providers/pluginfw/resources/user/data_users.go

```diff
@@ -0,0 +1,113 @@
+package user
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+func DataSourceUsers() datasource.DataSource {
+	return &UsersDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &UsersDataSource{}
+
+type UsersDataSource struct {
+	Client *common.DatabricksClient
+}
+
+type UsersInfo struct {
+	DisplayNameContains string     `json:"display_name_contains,omitempty" tf:"computed"`
+	UserNameContains    string     `json:"user_name_contains,omitempty" tf:"computed"`
```

Comment:
This is a fairly prescriptive interface: it basically says you can search for users based on whether their display name or user name contains a certain string. However, the REST API permits much more flexibility, allowing users to specify a custom filter where they could check if the username contains multiple strings, for example. I don't think we need to introduce a new abstraction & validation specifically for Terraform. What do you think about exposing a Filter field:
```go
type usersInfo struct {
        Filter types.String `json:"filter,omitempty"`
        ...
}
```
This one should not be computed, and we can pass it directly to the API request.